payment-request/allowpaymentrequest/removing-allowpaymentrequest.https.sub.html

```diff
@@ -13,26 +13,49 @@
   const path = location.pathname.substring(0, location.pathname.lastIndexOf('/') + 1);
   iframe.src = "https://{{domains[www1]}}:{{ports[https][0]}}" + path + "echo-PaymentRequest.html";
   iframe.onload = t.step_func(() => {
-    if (i === 1) {
+    if (i === 0) {
+      // 2. The iframe has now loaded (the first load).
+      // The allowpaymentrequest flag is set.
       iframe.allowPaymentRequest = false;
+      // 3. The allowpaymentrequest attribute has now been removed.
+      // The allowpaymentrequest flag is *still set* for the document.
     }
+    // 4. (first load) Ask the subdocument to invoke PaymentRequest and post back the result.
+    // 7. (second load) Ask the subdocument to invoke PaymentRequest and post back the result.
     iframe.contentWindow.postMessage('What is the result of new PaymentRequest(...)?', '*');
   });
 
   window.onmessage = t.step_func((e) => {
     i++;
     if (i === 1) {
+      // 5. This is the first message we receive, from the first load.
```

Comment:
In the future, we should write these as a promised-message queue. Would avoid having to keep a step index.